### PR TITLE
Adapts ZEP2 to codec

### PR DESCRIPTION
**Pull request by normanrz**
Makes changes to the current ZEP2 draft to switch sharding from the storage transformer model to a codec.

### CODE REVIEW

**Review by jstriebel — APPROVED** [review]
LGTM, @jbms could you have a look as well?

File: draft/ZEP0002.md [hunk 1]
```diff
@@ -138,107 +134,151 @@ chunks as sharding units, as it combines multiple chunks that are close to each
...
+            }
+          }
+        ]
+      }
+    }
+  ]
+}
+```
+
+#### `chunk_grid`
```
Comment (jbms) [inline draft/ZEP0002.md:203]: Should presumably be changed back to just `chunk_shape` now

**Comment by jstriebel** [comment]
@joshmoore @MSanKeys963 @rabernat Could one of you merge this PR? This also reflects the changes from https://github.com/zarr-developers/zarr-specs/pull/223

**Comment by MSanKeys963** [comment]
Thanks for sending the PR, @normanrz.
Merging this now.

**Pull request merged**